src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,901 @@
+            Assert.True(info.LastAccessTime > StartTime)
+            Assert.True(info.Name = IO.Path.GetFileName(TestBase.TestDirectory))
+            Assert.True(TestBase.RemoveEndingSeparator(info.Parent.ToString) = TestBase.RemoveEndingSeparator(IO.Path.GetTempPath()))
+            Assert.True(info.Root.Name = IO.Path.GetPathRoot(TestBase.TestDirectory))
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub GetDriveInfo_Drive()
+            Dim TestBase As New FileIOTestBase
+            Dim Rootname As String = FileSystem.GetDirectoryInfo(TestBase.TestDirectory).Root.Name
+            Dim Info As IO.DriveInfo = FileSystem.GetDriveInfo(Rootname)
+            Assert.NotNull(Info)
+            Dim DriveInfoFromSystemIO As IO.DriveInfo = Nothing
+
+            For Each Drive As IO.DriveInfo In IO.DriveInfo.GetDrives()
+                If Drive.Name = Rootname Then
+                    DriveInfoFromSystemIO = Drive
+                End If
+            Next
+            Assert.NotNull(DriveInfoFromSystemIO)
+
+            With DriveInfoFromSystemIO
+                Assert.True(Info.DriveFormat = .DriveFormat)
+                Assert.True(Info.DriveType = .DriveType)
+                Assert.True(Info.Name = .Name)
+                Assert.True(Info.RootDirectory.Name = .RootDirectory.Name)
+                Assert.True(Info.VolumeLabel = .VolumeLabel)
+            End With
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub GetFileInfo_File()
+            Dim StartTime As Date = Now
+            Threading.Thread.Sleep(1000)
+
+            Dim TestBase As New FileIOTestBase
+            Dim TestFile As String = CreateTestFile(TestBase, SourceData)
+
+            Dim FileInfoFromSystemIO As IO.FileInfo = New IO.FileInfo(TestFile)
+            Assert.NotNull(FileInfoFromSystemIO)
+
+            Dim info As IO.FileInfo = FileSystem.GetFileInfo(TestFile)
+            Assert.NotNull(info)
+            With FileInfoFromSystemIO
+                Assert.True(info.Exists)
+                Assert.True(info.Attributes = .Attributes)
+                Assert.True(info.CreationTime > StartTime)
+                Assert.True(info.DirectoryName = .DirectoryName)
+                Assert.True(info.Extension = .Extension)
+                Assert.True(info.FullName = .FullName)
+                Assert.True(info.IsReadOnly = .IsReadOnly)
+                Assert.True(info.LastAccessTime = .LastAccessTime)
+                Assert.True(info.LastWriteTime = .LastWriteTime)
+                Assert.True(info.Length = .Length)
+                Assert.True(info.Name = .Name)
+            End With
+
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub GetFiles_Directory()
+            Dim TestBase As New FileIOTestBase
+            Dim FileList As ReadOnlyCollection(Of String) = FileSystem.GetFiles(TestBase.TestDirectory)
```

Comment:
create a subdirectory and perform tests on that

Reply:
TestDirectory is a new Subdirectory for this test only.